cap/modules/mail/attributes.py

```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Analysis Preservation Framework.
+# Copyright (C) 2021 CERN.
+#
+# CERN Analysis Preservation Framework is free software; you can redistribute
+# it and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Analysis Preservation Framework is distributed in the hope that it will
+# be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Analysis Preservation Framework; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+from flask import current_app
+
+from .custom import body as custom_body
+from .custom import recipients as custom_recipients
+from .custom import subject as custom_subjects
+
+from .conditions import CONDITION_METHODS
+from .utils import EMAIL_REGEX, get_config_default, \
+    populate_template_from_ctx, update_mail_list
+
+
+def check_condition(record, condition):
+    """
+    For each condition we have a group of checks to perform, which should
+    return True/False
+
+    Each condition has the following:
+    - `checks`: the checks to be evaluated
+    - `op`: the operation to be used for all the checks (and/or)
+    - `mails`: the mail list
+
+    Each check has their own attributes:
+    - `if`: the type of check (e.g. exists, is_in, etc)
+    - `value`: the result
+    - `path`: the path that gets the field to be evaluated
+
+    An example of conditions:
+    {
+      "checks": [{
+        "path": "parton_distribution_functions",
+        "if": "exists",
+        "value": true
+      }],
+      "mails": {
+        "default": ["pdf-forum-placeholder@cern.ch"]
+      }
+    }
+
+    Nested checks are also allowed.
+    """
+    # for each condition we have a group of checks to perform
+    # all of the checks should give a True/False result
+    operator = condition.get('op', 'and')
+    checks = condition['checks']
+    check_results = []
+
+    for check in checks:
+        # get the method to apply, and use it on the required path/value
+        # for malformed conditions, we assume False
+        if check.get('checks'):
+            check_results.append(
+                check_condition(record, check)
+            )
+        elif check.get('if'):
+            try:
+                method = CONDITION_METHODS[check['if']]
+                check_results.append(
+                    method(record, check.get('path'), check.get('value', True))
```

Comment:
What if method fails? lets wrap with try/except and append `False` if something happens